lib/tools/gitignoreScanner.js

```diff
@@ -0,0 +1,41 @@
+// Module for scanning the .gitignore and constructing a list of files to ignore
+const fs = require('fs')
+const path = require('path')
+
+module.exports = function (gitignorePath) {
+  let fileList = ['lib-cov', 'node_modules', 'pids', 'logs', 'results', 'Thumbs.db', '*.seed', 'npm-debug.log', '.build', 'public', '*.log', '*.csv', '*.dat', '*.out', '*.pid', '*.gz', '.DS_Store']
+  let gitData
+  let polishedList = []
+
+  try {
+    // ensure that we can access .gitignore
+    fs.accessSync(gitignorePath)
+  } catch (e) {
+    // if we can not, return list of defaults
+    return fileList
+  }
+
+  // get data from .gitignore, trim it of whitespace, split at new lines
+  gitData = fs.readFileSync(gitignorePath).toString().split('\n')
```

Comment:
1. why is `toString()` necessary? shouldn't `readFileSync` give you a string already?
2. does this account for differences in how each OS handles line endings? what if the .gitignore file is authored in a windows environment?

Reply:
For point 2 I think it would be safer to use `.split(/\r?\n/)` to ensure OS compatibility.

Reply:
For point 1, just using `readFileSync` returns a buffer string to gitData like:
 `<Buffer 6c 69 62 2d 63 6f 76 0a 2a 2e 73 65 65 64 0a 2a 2e 6c 6f 67 0a 2a 2e 63 73 76 0a 2a 2e 64 61 74 0a 2a 2e 6f 75 74 0a 2a 2e 70 69 64 0a 2a 2e 67 7a 0a ... >`
 Then after appending a `toString()` to it, gitData will get returned the content of the gitignore file like we need